### PR TITLE
Pass predefined blocks via the CLI

### PR DESCRIPTION
**Pull request by xgreenx**
Follow-up PR for https://github.com/FuelLabs/fuel-core/pull/2081.

Closes https://github.com/FuelLabs/fuel-core/issues/1902

Add a CLI argument to process predefined blocks.

## Checklist
- [x] New behavior is reflected in tests

### Before requesting review
- [x] I have reviewed the code myself

### CODE REVIEW

File: tests/test-helpers/src/lib.rs [hunk 1]
```diff
@@ -9,3 +25,29 @@ pub async fn send_graph_ql_query(url: &str, query: &str) -> String {
...
+            Default::default(),
+        )
+        .add_output(Output::change(
+            Default::default(),
+            Default::default(),
+            Default::default(),
+        ))
+        .finalize_as_transaction();
+    let status = client
+        .submit_and_await_commit(&contract_tx)
```
Comment (MitchTurner) [inline tests/test-helpers/src/lib.rs:46]: [nit]:
```suggestion
    let script_tx = TransactionBuilder::script(vec![], vec![])
        .add_unsigned_coin_input(
            secret,
            rng.gen(),
            1234,
            Default::default(),
            Default::default(),
        )
        .add_output(Output::change(
            Default::default(),
            Default::default(),
            Default::default(),
        ))
        .finalize_as_transaction();
    let status = client
        .submit_and_await_commit(&script_tx)
```

**Review by MitchTurner — COMMENTED** [review]
Looking good.

File: crates/fuel-core/src/combined_database.rs [hunk 2]
```diff
@@ -264,44 +265,35 @@ impl CombinedDatabase {
...
                 return Err(anyhow::anyhow!(
-                    "gas-price-chain database height is less than target height"
+                    "gas-price-chain database height({gas_price_chain_height}) \
+                    is less than target height({target_block_height})"
                 ));
             }
 
             if on_chain_height > target_block_height {
                 self.on_chain().rollback_last_block()?;
-                tracing::info!(
```
Comment (xgreenx) [inline crates/fuel-core/src/combined_database.rs:285]: We have the same log inside of the  `rollback_last_block`

File: crates/fuel-core/src/state/historical_rocksdb.rs [hunk 3]
```diff
@@ -265,15 +269,21 @@ where
             .next()
             .ok_or(DatabaseError::ReachedEndOfHistory)??;
 
+        self.rollback_block_at(latest_height)?;
+
+        Ok(latest_height)
+    }
+
+    fn rollback_block_at(&self, height_to_rollback: u64) -> StorageResult<()> {
```
Comment (MitchTurner) [inline crates/fuel-core/src/state/historical_rocksdb.rs:277]: I think I'd prefer the name
```suggestion
    fn rollback_block_to(&self, height: u64) -> StorageResult<()> {
```

File: crates/fuel-core/src/service/adapters/graphql_api/off_chain.rs [hunk 4]
```diff
@@ -177,7 +177,7 @@ impl worker::OffChainDatabase for Database<OffChain> {
     type Transaction<'a> = StorageTransaction<&'a mut Self> where Self: 'a;
 
     fn latest_height(&self) -> StorageResult<Option<BlockHeight>> {
-        self.latest_height()
+        Ok(fuel_core_storage::transactional::HistoricalView::latest_height(self))
```
Comment (MitchTurner) [inline crates/fuel-core/src/service/adapters/graphql_api/off_chain.rs:180]: Let's use normal method syntax.

File: crates/fuel-core/src/service/adapters/graphql_api/on_chain.rs [hunk 5]
```diff
@@ -140,6 +140,6 @@ impl OnChainDatabase for OnChainIterableKeyValueView {}
 
 impl worker::OnChainDatabase for Database<OnChain> {
     fn latest_height(&self) -> StorageResult<Option<BlockHeight>> {
-        self.latest_height()
+        Ok(fuel_core_storage::transactional::HistoricalView::latest_height(self))
```
Comment (MitchTurner) [inline crates/fuel-core/src/service/adapters/graphql_api/on_chain.rs:143]: Let's use normal method syntax.

File: crates/fuel-core/src/state.rs [hunk 6]
```diff
@@ -71,7 +71,7 @@ pub trait TransactableStorage<Height>: IterableStore + Debug + Send + Sync {
 
     fn latest_view(&self) -> StorageResult<IterableKeyValueView<Self::Column>>;
 
-    fn rollback_last_block(&self) -> StorageResult<()>;
+    fn rollback_block_at(&self, height: &Height) -> StorageResult<()>;
```
Comment (MitchTurner) [inline crates/fuel-core/src/state.rs:74]: I think I prefer
```suggestion
    fn rollback_block_to(&self, height: &Height) -> StorageResult<()>;
```

**Review by MitchTurner — APPROVED** [review]

**Pull request merged**